Storage/Storage/Bookmarks.swift

```diff
@@ -34,14 +36,11 @@ public struct BookmarkRoots {
 /**
  * The immutable base interface for bookmarks and folders.
  */
-@objc public protocol BookmarkNode {
-    var guid: String { get }
-    var title: String { get }
-    var icon: UIImage { get }
-}
-
-@objc public protocol Bookmark : BookmarkNode {
-    var url: String! { get }
+public class BookmarkNode {
+    public var id: Int? = nil
+    public var guid: String = ""
+    public var title: String = ""
```

Comment:
I removed the default constructor here. You always have to pass in a value for the title of a bookmark to the constructor (i.e. implicit unwrapping is evil).